cmd/fleetctl/query.go

```diff
@@ -123,6 +123,49 @@ func queryCommand() *cli.Command {
 				return errors.New("Query must be specified with --query or --query-name")
 			}
 
+			labels := strings.Split(flLabels, ",")
+			if len(flLabels) > 0 {
+				labelSpecs, err := client.GetLabels()
+				if err != nil {
+					return fmt.Errorf("could not list labels: %w", err)
+				}
+
+				labelsFound := make(map[string]bool)
+				// create a map of labels to check
+				for _, label := range labels {
+					labelsFound[strings.TrimSpace(label)] = false
+				}
+				// iterate through labels available from the platform and mark the ones that are found
+				for _, label := range labelSpecs {
+					if _, ok := labelsFound[label.Name]; ok {
+						labelsFound[label.Name] = true
+					}
+				}
+				// generate list of labels that were not found
+				labelsNotFound := make([]string, 0, len(labels))
+				for label, found := range labelsFound {
+					if !found {
+						labelsNotFound = append(labelsNotFound, label)
+					}
+				}
+
+				if len(labelsNotFound) > 0 {
+					pluralizedLabel := "label"
+					if len(labelsNotFound) > 1 {
+						pluralizedLabel += string("s were")
+					} else {
+						pluralizedLabel += string(" was")
+					}
+
+					return errors.New(
+						fmt.Sprintf("The %s not found: %s. Please remove from --labels to continue.",
+							pluralizedLabel,
+							strings.Join(labelsNotFound, ", "),
+						),
+					)
+				}
+			}
```

Comment:
It doesn't appear that we use this `--labels` flag anywhere else, but if I am missing a location, perhaps it is worth to extract this validation out and apply in other locations.